sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedSource.java

```diff
@@ -133,6 +133,95 @@ public Double getFractionConsumed() {
     }
 
     /**
+     * A constant to use as the return value for {@link #getParallelismConsumed()} or
+     * {@link #getParallelismRemaining()} when the exact value is unknown.
+     */
+    public static final long PARALLELISM_UNKNOWN = -1;
```

Comment:
We can... but since the caller already has to check that the value is non-negative (and interpret that as UNKNOWN) it seemed simpler to include that behavior naturally. Neither way is very pretty, but java doesn't have unsigned values.